### PR TITLE
Fixed menu callback, only print picks if they are there

### LINKED ISSUE #1050: staff pick campaigns - Mobile Commons id/MailChimp group subscriptions
http://staging.beta.dosomething.org/admin/config/services/optins is returning an access denied right now; need to be able to access it!

bonus points if Marah and Alysha can access it without giving them full admin access (but if that doesn't happen for this release it's fine)

relates to #859 and #893

### PR DESCRIPTION
**Pull request by angaither**
Fixes #1050

:airplane: :clock1:

### CODE REVIEW

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.admin.inc [hunk 1]
```diff
@@ -26,28 +26,30 @@ function dosomething_signup_optin_config($form, &$form_state) {
...
-    );
+      // Each campagin gets a mailchimp & mobile commons id.
+      $form[$nid]['dosomething_mailchimp_id_' . $nid] = array(
+        '#type' => 'textfield',
+        '#title' => t('MailChimp Group'),
+        '#default_value' => variable_get('dosomething_mailchimp_id_' . $nid, ''),
+      );
+      $form[$nid]['dosomething_mobilecommons_id_' . $nid] = array(
+        '#type' => 'textfield',
+        '#title' => t('Mobile Commpns Opt-In'),
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.admin.inc]: Typo.

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.admin.inc [hunk 2]
```diff
@@ -26,28 +26,30 @@ function dosomething_signup_optin_config($form, &$form_state) {
...
-    $form[$nid]['dosomething_mobilecommons_id_' . $nid] = array(
-      '#type' => 'textfield',
-      '#title' => t('Mobile Commpns Opt-In'),
-      '#default_value' => variable_get('dosomething_mobilecommons_id_' . $nid, ''),
-    );
+      // Each campagin gets a mailchimp & mobile commons id.
+      $form[$nid]['dosomething_mailchimp_id_' . $nid] = array(
+        '#type' => 'textfield',
+        '#title' => t('MailChimp Group'),
+        '#default_value' => variable_get('dosomething_mailchimp_id_' . $nid, ''),
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.admin.inc]: I recently had to set up variables for campaigns nodes... and went with a naming convention like `dosomething_campaign_nid_723_alt_color`.  I think it'd be useful to specify nid in the variable names so there's no confusion.  What do you think of using names like `dosomething_signup_nid_723_mobilecommons_optin` ?

**Comment by angaither** [comment]
@aaronschachter updated. no issues adding `nid` to the var name, I find them easier to read when the variable is at the end of the name.

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.admin.inc [hunk 3]
```diff
@@ -26,28 +26,30 @@ function dosomething_signup_optin_config($form, &$form_state) {
...
-      '#title' => t('MailChimp Group'),
-      '#default_value' => variable_get('dosomething_mailchimp_id_' . $nid, ''),
-    );
-    $form[$nid]['dosomething_mobilecommons_id_' . $nid] = array(
-      '#type' => 'textfield',
-      '#title' => t('Mobile Commpns Opt-In'),
-      '#default_value' => variable_get('dosomething_mobilecommons_id_' . $nid, ''),
-    );
+      // Each campagin gets a mailchimp & mobile commons id.
+      $form[$nid]['dosomething_mailchimp_id_' . $nid] = array(
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.admin.inc]: These guys need the `nid` prefix added too in order for system settings form to work.

**Comment by angaither** [comment]
@aaronschachter gah, try again!

**Comment by aaronschachter** [comment]
:+1: ship it value added

**Pull request merged**